src/Trinity.C/src/Storage/MemoryTrunk/MemoryTrunk.Memory.cpp

```diff
@@ -250,8 +263,7 @@ namespace Storage
         size_t append_head = cp - buffer;
         if (Memory::MemoryCommit(trunkPtr, append_head) == NULL)
         {
-            free(buffer);
-            return -1;
+            Trinity::Diagnostics::FatalError("ReloadImpl: MemoryTrunk {0} fails to reclaim memory after data's been moved to a temporary buffer.", TrunkId);
```

Comment:
@leoxia this is a fatal error: the trunk cannot be created after the data has been dumped into buffer to reload. if we return here, the data is gone, and any further access to this trunk causes access violation. Trinity::Diagnostics::FatalError implies exit()